### DISCUSSION TITLE
[Desglose][WIP] Power Ups

### DISCUSSION THREAD

**Discussion by emrodk** [Godot-en-Espanol/Node-City#16] [Tareas]
Work In Progress (WIP)

- TODO Transformar las poderes de las estrellas del juego original a power up único

## Requisitos
 - Que requiera un poco de skills. No debe ser muy OP. Ej. la granada pierda gracia, debería matar tanques en un área reducida
 - Pueden ser activables

# Power Ups

## Granada 
| ![Battle_City_Powerup_Grenade](https://user-images.githubusercontent.com/51797625/120110429-c8c11a00-c143-11eb-900f-d2755d1b7df4.png) | Destruye a todos los enemigos en pantalla. Hay penalización de puntaje |
|:-:|:-:|
* Pros:
  * Si la situación se vuelve muy difícil, el jugador puede ganar una chance más
* Contras
  * Si el jugador no requería usar la granada para ganar la ola esto afecta de manera negativa al score

## Casco
| ![Battle_City_Powerup_Helmet](https://user-images.githubusercontent.com/51797625/120110428-c8288380-c143-11eb-819e-58a491067796.png) | Campo de fuerza temporal similar a cuando el PJ spawnea. (Inmunidad) |
|-|-|
| Pros | |
| Contras | |

## Pala
| ![Battle_City_Powerup_Shovel](https://user-images.githubusercontent.com/51797625/120110427-c8288380-c143-11eb-90a3-657438bf1a8c.png)|  Convierte de manera temporal las paredes de ladrillos en paredes metálicas alrededor de la base. Si las mismas están rotas las repara |
|-|-|
| Pros | |
| Contras | |

## Estrella
| ![Battle_City_Powerup_Star](https://user-images.githubusercontent.com/51797625/120110426-c78fed00-c143-11eb-9325-db6ba7b22c1f.png)| Aumenta el poder de disparo un nivel |
|-|-|
| Pros | |
| Contras | |

## Tanque
| ![Battle_City_Powerup_Tank](https://user-images.githubusercontent.com/51797625/120110425-c6f75680-c143-11eb-834e-7df8d3e2add1.png) | Te da una vida extra. (También se puede conseguir una vida si se llega los 20.000 puntos)|
|-|-|
| Pros | |
| Contras | |

## Temporizador
| ![Battle_City_Powerup_Timer](https://user-images.githubusercontent.com/51797625/120110368-8992c900-c143-11eb-8cf0-c61607d8c6c3.png)| Congela el tiempo, deteniendo el movimiento de todos los enemigos.|
|-|-|
| Pros | |
| Contras | |

## Decoy Activable
* No invisibilidad, la AI sigue persiguiendo al player cercana

**Comment by LeandroCuello** [comment]
En mi opinión todas las habilidades deberían ser activables. Creo que puede agregarle una capa mas de complejidad sin mucha dificultad y hacer que el juego resalte un poco mas comparado al original.
La idea seria que el jugador puede tomar una habilidad y se la quedaría hasta usarla o tomar otra, Osea solo puede mantener una a la vez. No tengo muy decidido de donde o como aparecerían las habilidades y no se si esto se debería definir en esta discusión.

Si entiendo bien la consigna tenemos que proponer nuestras ideas para las habilidades y lo haré a continuación, si no era para esto.. perdón por el spam jajaja.

**Granada**
La granada la imagino como un proyectil que se tira de forma parabólica, y cae a una determina distancia de donde apunta el tanque. Al caer eliminaría todos los tanques que se encuentre en el área donde ocurre la explosiona.
También podría ser que rompa el bloque donde callo la granada. Esto puede estar bueno para romper bloques con una dureza que aun no podes romper. También agrega el peligro de dejar descubierta la base o destruirla.

**Casco**
El casco no cambia mucho la forma de Jugar y no resulta muy divertido. Se me ocurre que funcione un poco como las estrella en los juegos de Mario; El tanque es inmune, no puede disparar y mata a contacto. Para hacer esto un poco mas divertido se me ocurría que podría tener un aumento de velocidad y reducir el tiempo de duración.
Otra opción es hacer que sea inmune, pueda disparar, se mueva mas lento y rompa paredes. Aunque esta opción tal vez genere que el mapa quede “pelado”.

**Pala**
Creo que la pala no pude funcionar igual si se hacen las habilidades como activas. Me imagino que los jugadores van a querer guardarse la habilidad hasta que algún taque este por destruir la base y ahí usarla. Esto genera que el jugador no quiera usar las otra habilidades porque la pala seria muy fuerte.
No tengo mucha propuesta para esta. Me gusto lo que que dijo Ema de que construya un bloque adelante o algo parecido.

**Tanque**
Bueno no tiene sentido que sea una activable. Tampoco los clasificaría como “powe up”. En mi opinión se tendría que charlar en otra discusión. Digo esto porque tal vez es mejor dar una vida extra por pasar un nivel, matar 𝑥 cantidad en enemigos en un cierto tiempo, no se… jajaja.

**Temporizador**
Emmmm….  Se hablo de un área que ralentizaría todo lo que entra… no se me ocurre nada che.
Decoy/invisibilidad
Me gusta la idea del decoy para jugar con la inteligencia de los bots. Creo que podrían funcionar con cierta proximidad. También creo que deberían quedarse un tiempo ilimitado hasta ser rotos. La invisibilidad también puede estar buena, pero no me gusta tanto.

**Estrella**
Se hablo de muchas opciones para la estrella en la reunión, vengo a proponer una mas. La idea seria que las mejoras sean parecidas a la del juego original. La diferencia recae en como se obtiene.
Un tanque dorado que no dispara pero es rápido aparecerá de manera aleatoria. Este se mueve por el mapa de manera errática. Si transcurre un tiempo sin ser destruido desparece. La idea seria que el jugador tenga que perseguir este tanque y cuando lo destruye obtiene la mejora.

**Comment by FROSADO** [comment]
**Granada** : una bomba en parábola que hace daño de area. El impacto directo hace como 3 disparos, el area 1. 

* pros : Daño masivo por area y rompe paredes. 
* Cons: Corto alcance, puede despejar demasiado la pantalla, la onda expansiva puede hacer daño al jugador y destruye otros powerups. 

**Casco**: Yo lo pondria como un escudo que soporta un disparo nada más. Es como incrementar la vida en un "corazón" de forma temporal. No te hace invulnerable, simplemente resistes un disparo más. 
* Pro: Da un poco de aire al jugador, permite hacer jugadas un poco más suicidas sabiendo que puede aguantar un tiro más. 
* Cons: No compatible con otros powerups. Falsa sensación de seguridad.

**Pala**: Una habilidad que debe de durar solo dos o tres bloques, con el Powerup solo puedes levantar dos muros, tres a lo sumo.  O reparar un muro parcialmente destruido. No veo correcto convertir un muro en otro tipo. Yo le pondria, como a todos los powerups, un tiempo de vida, si no lo usas en x min, desaparece, y que no permanezca al cambiar de nivel.
* Pro: Puedes mejorar la defensa de la bandera o cubrirte una retirada.
* Cons: Es lento levantar un muro, durante ese tiempo estas expuesto. Si tarda 1 / 2 segundos en levantar el muro, es un segundo sin poder moverte.

**Estrella**: Como dice  @LeandroCuello para el casco , lo pondría como la estrella del mario. invulnerable y destruye al toque los enemigos. No rompería muros.  Corta duración y el tanque se acelera. 
* PRO: Brilla y corre más. Los brillos siempre hacen la vida mejor. 
* CONS: Se acaba y puedes estar en medio de un grupo de enemigos a los que no ha dado tiempo a destruir. 

**Tanque**: Si vamos a hacer que el juego tenga vidas o "corazones", este sería el incremento de vidas o recuperar un corazón. No le daria más valor que eso. Cada cierto tiempo aparece como extra un "tanque" que debes coger para recuperar vida.  
No estoy a favor de numero de vidas. La filosofia de vidas y continuar de las recreativas prefiero discutirlo a parte. Pero es algo que no aporta nada, tener vida y luego dar la opción de continuar. 

* PRO: Puedes recuperarte de un error , recuperas un corazón de vida. 
* CONS: Si aparece con demasiada facilidad no hay miedo a hacer locuras. 
 

**Temporizador**: Si hacemos que las oleadas vayan por tiempo, aumenta el tiempo hasta la siguiente oleada. Mi idea es que cada x minutos se inicia una oleada, si en ese tiempo has terminado con todos los enemigos, mejor para el jugador, pero si no, se puede encontrar con todos los que no ha terminado de la oleada anterior y los nuevos de esta oleada. El temporizador retrasaría cuando va a iniciarse la siguiente oleada, dando más tiempo al jugador a terminar con la actual. 
* PRO: el jugador tiene más tiempo hasta que surja la siguiente oleada. 
* CONS: Tiempo sin hacer nada si no hay enemigos.
